scipy/ndimage/src/nd_image.c

```diff
@@ -45,6 +45,10 @@
 #include "ccallback.h"
 #include "numpy/npy_3kcompat.h"
 
+#if NPY_API_VERSION >= 0x0000000c
+    #define HAVE_WRITEBACKIFCOPY
+#endif
+
```

Comment:
I don't think this really changes much. Rather, I think the idea is to define a macro here. Probably something like this:

```C
#define HAVE_WRITEBACKIFCOPY(output)  if ( NPY_API_VERSION >= 0x0000000c ) \
                                         PyArray_ResolveWritebackIfCopy(output);
```

And then just a single `HAVE_WRITEBACKIFCOPY(output)` statement should suffice at the various locations where there are the `ifdef` blocks I think. Just have to be careful with the preprocessor literal substitutions (my syntax might be slightly off).

Reply:
That's not going to work, as the symbol will not be resolved at compile-time.